src/http/ngx_http_wasm_api.c

```diff
@@ -402,20 +430,9 @@ proxy_add_header_map_value(int32_t type, int32_t key_data, int32_t key_size,
     char                       *key, *val;
 
     log = ngx_http_wasm_get_log();
-    r = ngx_http_wasm_get_req();
-    if (r == NULL) {
-        return PROXY_RESULT_BAD_ARGUMENT;
-    }
-
-    key = (char *) ngx_wasm_vm.get_memory(log, key_data, key_size);
-    if (key == NULL) {
-        return PROXY_RESULT_INVALID_MEMORY_ACCESS;
-    }
-
-    val = (char *) ngx_wasm_vm.get_memory(log, data, size);
-    if (val == NULL) {
-        return PROXY_RESULT_INVALID_MEMORY_ACCESS;
-    }
+    must_get_req(r);
```

Comment:
I prefer the old style, which is clearer.

Reply:
The old style has too much repeated code. It makes the real logic hard to recognize.